File: Queue/10845.cpp
#include <iostream>
#include <queue>
#include <string>

using namespace std;

int main() {
    int N;
    cin >> N;
    queue<int> q;
    
    while (N--) {
        string inst;
        int num;
        cin >> inst;
        
        if (inst == "push") {
            cin >> num;
            q.push(num);
        }
        else if (inst == "pop") {
            if (q.empty()) {
                cout << -1 << endl;
            }
            else {
                cout << q.front() << endl;
                q.pop();
            }
        }
        else if (inst == "size") {
            cout << q.size() << endl;
        }
        else if (inst == "empty") {
            if (q.empty()) {
                cout << 1 << endl;
            }
            else {
                cout << 0 << endl;
            }
        }
        else if (inst == "front") {
            if (q.empty()) {
                cout << -1 << endl;
            }
            else {
                cout << q.front() << endl;
            }
        }
        else if (inst == "back") {
            if (q.empty()) {
                cout << -1 << endl;
            }
            else {
                cout << q.back() << endl;
            }
        }
    }
}

File: Data-Structure/Queue/10845.py
import sys

N = int(input())

queue = list()

for i in range(N):
    cmd = sys.stdin.readline().split()

    if cmd[0] == 'push':
        queue.append(int(cmd[1]))

    elif cmd[0] == 'pop':
        if not queue:
            print('-1')
        else:
            print(queue[0])
            queue.pop(0)

    elif cmd[0] == 'size':
        print(len(queue))

    elif cmd[0] == 'empty':
        if not queue:
            print('1')
        else:
            print('0')

    elif cmd[0] == 'back':
        if not queue:
            print('-1')
        else:
            print(queue[-1])

    elif cmd[0] == 'front':
        if not queue:
            print('-1')
        else:
            print(queue[0])
